webui/react/src/shared/hooks/useModal/useModal.tsx

```diff
@@ -20,6 +20,12 @@ export interface ModalHooks {
   modalRef: React.MutableRefObject<ReturnType<ModalFunc> | undefined>;
 }
 
+export type ModalOpen<T> = (modalProps?: ModalFuncProps & { openProps?: T }) => void;
```

Comment:
👀

Reply:
I thought about this during the modal refactor and I didn't take the time to do this then, but what do you think about having `modalOpen` just take in a 2nd param of custom props/context?
```
modalOpen: (modalProps?: ModalFuncProps = {}, context?: RecordUnknown) => void;
```

Reply:
and you're thinking the `context` would be used the same way openProps is used? should we still ask for the user to defien the type instead of `RecordUnknown` we could default to RecordUnknown instead of just `unknown`

Reply:
if so that works but I wouldn't see much of a difference. looping in @johnkim-det as well since we had some discussion on this [here](https://github.com/determined-ai/saas/pull/205)

Reply:
yea we considered having two params. personally i preferred not having to pass undefined or empty object if there are no 'modalProps' being used.

Reply:
hmmm good points. My thought is to achieve the following:
1. Preserve the modalProps as something that continuously gets exposed during composition.
2. allow additional context to be passed in for the newly composed modal hook.

What about something something like this?
```
interface ModalProps extends ModalFuncProps {
  context?: RecordUnknown;
}

...
modalOpen: (modalProps: ModalProps = {}) => void;
...
```

Reply:
sounds good to me. 
to clarify the changes:
this is the same as we have here with the differences being:
- renaming the key to `context` instead of `openProps`
- apply the change to the original `ModalHooks` instead of adding a new type

Reply:
yeah do those two changes sound ok to you two?